src/operators/suffix.ts

```diff
@@ -160,7 +160,13 @@ export class SuffixOperator implements Operator {
   }
 
   handleData(data: any[]): any[] | null {
-    const index = this.index >= 0 ? this.index : data.length + this.index;
+    let index = this.index >= 0 ? this.index : data.length + this.index;
+
+    // check if the `source` param was included and if so decrement the index
```

Comment:
I know this is already merged, but how is the source parameter provided?

Reply:
It's coming in from the `beforeDestination` setting.  So what's happening is that every rule is automatically getting the `suffix` operator run on it before the destination.  Where the real problem lies is souq, which is now adding the source parameter.  But since the suffix operated had defaulted to looking at the last arg (FS.event and FS.setUserVar both have objects as the last arg), this is a problem.  This "fix" essentially corrects what is more accurately a bug in souq, but it would guard against a rule write that for whatever reason also added the source param.

Reply:
Got - I leveled up a little :)

Would this have had the same desired outcome if the logic were

```
if (typeof data[index] !== 'object') {
      index -= 1;
    }
```

?

Reply:
It would.  It's also an edge case, but we'd have to delineate between a POJO and an array since the typeof for an array returns `object`.  If we want to adjust the check a bit, I could make that change.